internal/pkg/deploy/cloudformation/stack/app.go

```diff
@@ -111,6 +111,20 @@ func (c *AppStackConfig) ResourceTemplate(config *AppResourcesConfig) (string, e
 	return content.String(), err
 }
 
+func appTemplatePath(version string) string {
+	if version == "" {
```

Comment:
An probably unimportant question: I'm wondering when would `c.Version` be left empty, or it will be empty only by mistake? Do we want `c.Version` to always be non empty (thus when `version == ""` we want to be aware of it instead of defaulting to legacy template version)?

Reply:
It is empty when the current template is in legacy version because it won't have `templateVersion` in its metadata section. For all the following app template versions, we are supposed to be able to parse it to get the version number.